core/trino-main/src/main/java/io/trino/execution/SqlStageExecution.java

```diff
@@ -167,6 +171,7 @@ private SqlStageExecution(
             }
         }
         this.exchangeSources = fragmentToExchangeSource.build();
+        this.outboundDynamicFilterIds = getOutboundDynamicFilters(stateMachine.getFragment());
```

Comment:
use sepatate feature toggle for enabling sending of DFs to workers since it increases communication overhead number_of_worker times.